python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -138,9 +147,14 @@ def handle_output(self, context: OutputContext, obj: object) -> None:
 
     def load_input(self, context: InputContext) -> object:
         obj_type = context.dagster_type.typing_type
-        self._check_supported_type(obj_type)
+        if obj_type is Any and self._default_load_type is not None:
+            load_type = self._default_load_type
+        else:
+            load_type = obj_type
+
+        self._check_supported_type(load_type)
 
-        return self._handlers_by_type[obj_type].load_input(
+        return self._handlers_by_type[load_type].load_input(
```

Comment:
Is there a way that we could hint to the user that this is happening, in the case that it leads to unexpected behavior? I could see a situation where a user might be confused why their inputs are presented as a particular type.

I think this is a fairly minor issue especially since it only affects input loading, so not blocking (in theory there should be no issues with calling `load_input`, I could see default output types being more tricky)